chore(deps): bump storybook to 9.1.13 DEV-888

magicznyleszek opened the pull request:
### 🗒️ Checklist

1. [x] run linter locally
2. [x] update developer docs (API, README, inline, etc.), if any
3. [x] for user-facing doc changes create a Zulip thread at #Kobo support docs, if any
4. [x] draft PR with a title `<type>(<scope>)<!>: <title> DEV-1234`
5. [x] assign yourself, tag PR: at least `Front end` and/or `Back end` or `workflow`
6. [x] fill in the template below and delete template comments
7. [x] review thyself: read the diff and repro the preview as written
8. [x] open PR & confirm that CI passes & request reviewers, if needed
9. [ ] delete this section before merging

### 💭 Notes

Updated storybook packages versions with their internal tool. 

Also updated `react-refresh` and the other refresh package, as I thought it would fix a problem with `@storybook/addon-webpack5-compiler-swc`. It didn't. But the bump didn't break anything, so I left it there. 

The `TerserPlugin` configuration in `.storybook/main.js` was crashing `(npx) storybook build` command. I've changed it a bit following some advice online. AFAIK it is not using `swc` now, but it is still helpful :shrug: The previous work here was https://github.com/kobotoolbox/kpi/commit/19e5a5ff0c4360e336c84c5ffb9a98d2079ff78c

Added `@typescript-eslint/parser` because eslint complained without it. My guess is that update to `eslint-plugin-storybook` escalated this requirement :shrug:

### 👀 Preview steps

1. `npm install`
2. `npm run storybook`
3. go to http://localhost:6006/
4. 🟢 [on PR] notice that it still works

pauloamorimbr commented on package.json:
One of the issues is this missing 
```
    "@biomejs/cli-linux-x64": "^2.2.4",
```
on devDependencies

p2edwards reviewed (commented):
Still looking at this locally. On my current setup, the refresh plugin doesn't work when using watch mode (`npx storybook dev -p 6006`), and there's an error in the browser console related to the dark mode plugin.

magicznyleszek commented:
As per discussion on standup, @Akuukis will take over this one together with @p2edwards :) Thanks!

Akuukis committed:
chore(deps): bump storybook to 9.1.13 DEV-888

Squashed commit of the following:

commit 8835a109fe375f1b57cf3906964e05d9437cb838
Merge: 425c33a1b 3a1c8520c
Author: Leszek Pietrzak <leszek@magicznyleszek.xyz>
Date:   Mon Sep 22 16:03:21 2025 +0100

    Merge branch 'main' into leszek/bump-storybook-version

commit 425c33a1b2062b03cad819cb355a4bf22cb8db47
Author: Leszek <smutnyleszek@gmail.com>
Date:   Mon Sep 22 16:55:21 2025 +0200

    fix types in usage query

commit b6cab6eb232a8dfdf15c4a8968e2e94fde1dd913
Author: Leszek <smutnyleszek@gmail.com>
Date:   Mon Sep 22 14:34:35 2025 +0200

    try bumping tsconfig lib to es2022

commit 2b6a8ec1f5960c51b1b87457bf97a2a49666a61b
Author: Leszek <smutnyleszek@gmail.com>
Date:   Mon Sep 22 14:28:01 2025 +0200

    npm update rebuilt msw-mocks file :+1:

commit 29a00196ec37525d523bdd71fdedcc85171be06d
Author: Leszek <smutnyleszek@gmail.com>
Date:   Mon Sep 22 14:27:44 2025 +0200

    biome ignore package.json

commit a03e266c94484650bec3c2b340102f816f0d8349
Author: Leszek <smutnyleszek@gmail.com>
Date:   Mon Sep 22 13:56:56 2025 +0200

    delete unneeded patch

commit 595bee4fdb3deb52e2f06d6fe08f638feac611f5
Author: Leszek <smutnyleszek@gmail.com>
Date:   Mon Sep 22 13:47:39 2025 +0200

    rebuild package-lock with "npm update"

commit bd02dcc1112613cd49c827904b8129254192b289
Author: Leszek <smutnyleszek@gmail.com>
Date:   Mon Sep 22 13:40:19 2025 +0200

    run biome

commit a056520746d648566280b273b2c866a15210451b
Author: Leszek <smutnyleszek@gmail.com>
Date:   Mon Sep 22 10:51:22 2025 +0200

    biome fixes

commit e388567b83f638c180cedebff20605d5238b8c99
Author: Leszek <smutnyleszek@gmail.com>
Date:   Mon Sep 22 10:49:24 2025 +0200

    biome missing dep hack

commit 0e67d733bb5ed4e9807ba88d966b71e1ad11c286
Merge: 7d4d222bd db47424d1
Author: Leszek Pietrzak <leszek@magicznyleszek.xyz>
Date:   Fri Sep 19 11:57:22 2025 +0100

    Merge branch 'main' into leszek/bump-storybook-version

commit 7d4d222bd6fad123fed706eaed26195b9810a3e0
Merge: 60170b685 1cdd31faf
Author: Leszek <smutnyleszek@gmail.com>
Date:   Wed Sep 17 22:39:43 2025 +0200

    Merge branch 'main' into leszek/bump-storybook-version

commit 60170b6852ba3f3ef5fec31da7222f467d0509fe
Merge: fa60e3fca 3ca894632
Author: Leszek Pietrzak <leszek@magicznyleszek.xyz>
Date:   Mon Aug 18 10:16:57 2025 +0200

    Merge branch 'main' into leszek/bump-storybook-version

commit fa60e3fcad6a481a4d1ad07d6ef85c5070a8a3b4
Merge: f779d34ce e2d3329dd
Author: Leszek Pietrzak <leszek@magicznyleszek.xyz>
Date:   Thu Aug 14 10:59:26 2025 +0200

    Merge branch 'main' into leszek/bump-storybook-version

commit f779d34cea4407a16a6b459177c415f561f3ee11
Author: Leszek <smutnyleszek@gmail.com>
Date:   Tue Aug 12 20:41:45 2025 +0200

    add @typescript-eslint/parser

commit bcf2565ddb5a0c4ec929765a8f6795da8e4f98ab
Author: Leszek <smutnyleszek@gmail.com>
Date:   Tue Aug 12 20:04:29 2025 +0200

    add missing @storybook/types

commit 11b9748b24110e1e3ac6d9277b66f5c94c4ed203
Author: Leszek <smutnyleszek@gmail.com>
Date:   Tue Aug 12 19:15:11 2025 +0200

    remove old comment, bump versions again, chane terserplugin setup

commit 38ddb86cd9cb47cfd71496ede52c9565f0a109e1
Merge: 0cd36cb3a 9c7f6ec4e
Author: Leszek Pietrzak <leszek@magicznyleszek.xyz>
Date:   Tue Aug 12 17:52:39 2025 +0200

    Merge branch 'main' into leszek/bump-storybook-version

commit 0cd36cb3a46645e0c371c18011cd0c8fc29d1731
Merge: 43d9254ac 7d2dd7277
Author: Leszek Pietrzak <leszek@magicznyleszek.xyz>
Date:   Mon Aug 11 17:27:57 2025 +0200

    Merge branch 'main' into leszek/bump-storybook-version

commit 43d9254ac1c7fd1eb667f4e703eaf8d7bc37b9ff
Merge: 5c4843ff5 23f7f3466
Author: Leszek Pietrzak <leszek@magicznyleszek.xyz>
Date:   Mon Aug 11 11:50:16 2025 +0200

    Merge branch 'main' into leszek/bump-storybook-version

commit 5c4843ff557e9ed1606c6136a590feabb41f2d30
Merge: 687d527a8 47bc86f75
Author: Leszek Pietrzak <leszek@magicznyleszek.xyz>
Date:   Mon Aug 11 11:21:49 2025 +0200

    Merge branch 'main' into leszek/bump-storybook-version

commit 687d527a8a030aff1755028a25e73a575e006285
Author: Leszek <smutnyleszek@gmail.com>
Date:   Fri Aug 8 12:09:01 2025 +0200

    add missing package

commit ec6473dff2057bba4575a01312ea9d5d5fe66b69
Merge: 83c7394f3 5b407ec0c
Author: Leszek <smutnyleszek@gmail.com>
Date:   Fri Aug 8 12:02:13 2025 +0200

    Merge branch 'main' into leszek/bump-storybook-version

commit 83c7394f365ada7459af0295923b26721e41cb32
Author: Leszek <smutnyleszek@gmail.com>
Date:   Fri Aug 8 11:52:18 2025 +0200

    bump storybook version(s)

Akuukis commented:
Depends on (at least) bumping react-router first, let's put back to a draft this PR in the meantime.

magicznyleszek commented:
@Akuukis react-router got updated: https://github.com/kobotoolbox/kpi/pull/6979

magicznyleszek commented:
DEV-888 is "Won't fix" so I'm closing this
